internal/api/websocket_client.go

```diff
@@ -154,44 +164,40 @@ func (c *WebsocketClient) readPump() {
 	})
 
 	for {
-		_, message, err := c.conn.ReadMessage()
+		request, err := c.Receive()
 		if err != nil {
-			if websocket.IsUnexpectedCloseError(err, websocket.CloseGoingAway, websocket.CloseAbnormalClosure, websocket.CloseNoStatusReceived) {
-				c.logger.WithErr(err).Errorf("Unhandled websocket error")
+			if IsFatalStreamError(err) {
+				c.cancel()
+				break
 			}
-			c.cancel()
-			break
+
+			continue
 		}
 
-		if err := c.handle(message); err != nil {
+		if err := HandleStreamingMessage(c, request); err != nil {
 			c.logger.WithErr(err).Errorf("Handle websocket message")
 		}
 	}
 
 	close(c.stopCh)
 }
 
-func (c *WebsocketClient) handle(message []byte) error {
-	var request websocketRequest
-	if err := json.Unmarshal(message, &request); err != nil {
-		return err
-	}
-
-	handlers, ok := c.handlers[request.Type]
+func HandleStreamingMessage(client StreamingClient, request StreamRequest) error {
```

Comment:
Not a huge issue but for consistency, this method can be unexported just like `handleUnknownRequest`

Reply:
Addressed in the latest commit